### PR TITLE
unexport and undocument DT(), closes #5472

### LINKED ISSUE #5472: Do not export DT() yet
This is a new feature added in current dev version. There are multiple open issues related to it
https://github.com/Rdatatable/data.table/labels/DT%28%29
All those issues should be fixed before rolling out this new function. Therefore to not slow down 1.14.4 I suggest to mark DT() as not exported for the current moment, and re-export it shortly after release, so users of our dev pkg can still use it.

### PR DESCRIPTION
**Pull request by jangorecki**
closes #5472

Based on two PRs
https://github.com/Rdatatable/data.table/pull/5104
https://github.com/Rdatatable/data.table/pull/5113
functionality is still there, it is just not exported and documented.

I encourage users interested in this functionality to use
```r
DT = data.table:::DT
```
and play with it. Just be aware that there are couple open issue that affects DT() as of now, all listed in https://github.com/Rdatatable/data.table/labels/DT%28%29

### CODE REVIEW

**Comment by jangorecki** [comment]
inviting @eddelbuettel to review, as you mentioned that you use `DT()` function (in #5808)

**Review by eddelbuettel — COMMENTED** [review]
Not sure how much I can say here so I just leave a comment -- I got used to the _exported_ version that I use(d) in some workflows / experiments here.  I like its compactness, and forcing data.table::DT is much less attractive, as is having to make a local copy.

**Comment by MichaelChirico** [comment]
> I got used to the exported version

IINM the intention is to un-export it to remove it as a block to releasing 1.15.0, then export it again after fixing the pending issues. We might also export it in 1.15.99 dev version after a CRAN release.

File: inst/tests/tests.Rraw [hunk 1]
```diff
@@ -349,7 +351,7 @@ test(83, TESTDT[,list("a","b")], data.table(V1="a",V2="b"))
 test(83.1, TESTDT[,list("sum(a),sum(b)")], data.table("sum(a),sum(b)"))
 test(83.2, TESTDT[,list("sum(a),sum(b)"),by=a], {tt=data.table(a=c("a","c","d","g"),V1="sum(a),sum(b)",key="a");tt$V1=as.character(tt$V1);tt})
 test(84, TESTDT[1:2,list(a,b)], data.table(a=c("a","c"), b=c("e","e"), key = 'a,b'))
-# test(85, TESTDT[1:2,DT(a,b)], data.table(a=c("a","c"), b=c("e","e")))  #DT() now deprecated
+# test(85, TESTDT[1:2,DT(a,b)], data.table(a=c("a","c"), b=c("e","e")))  #DT() now deprecated ## this is very old DT() functionality, completely different than DT() discussed recently
```
Comment (MichaelChirico) [inline inst/tests/tests.Rraw:354]: "recently" will quickly become "when exactly", use a specific reference

**Comment by jangorecki** [comment]
Yes, the issue explains:
> Therefore to not slow down ~~1.14.4~~ 1.15.0 I suggest to mark DT() as not exported for the current moment, and re-export it shortly after release (if there will be such demand, see https://github.com/Rdatatable/data.table/issues/5621);

File: inst/tests/tests.Rraw [hunk 2]
```diff
@@ -8,6 +8,7 @@ if (exists("test.data.table", .GlobalEnv, inherits=FALSE)) {
   if ((tt<-compiler::enableJIT(-1))>0)
     cat("This is dev mode and JIT is enabled (level ", tt, ") so there will be a brief pause around the first test.\n", sep="")
   rm_all = function() {}
+  DTfun = DT ## otherwise DT would be re-defined by many tests
```
[the remark this answers is not in the document]
Comment (tdhock) [reply inst/tests/tests.Rraw:11]: agree

**Pull request merged**